### DISCUSSION TITLE
Accidentally deleted $USERHOME/nominatim-project

### DISCUSSION THREAD

**Discussion by RailsGuy** [osm-search/Nominatim#3563] [General]
I have a 4.5.0 instalI on ubuntu 22.04 and accidentally deleted the $USERHOME/nominatim-project directory after I had done a full planet, postcode, wiki etc import.

The ubuntu 22.04 install guide says the frontend needs $USERHOME/nominatim-project and it is populated during the import.
https://nominatim.org/release-docs/latest/admin/Install-on-Ubuntu-22/#setting-up-a-webserver

I have recreated the directory but it is of course empty.
How bad is this? 
What does the import write to $USERHOME/nominatim-project and can I re-create that content manually now based on my import?

**Comment by mtmail** [comment]
With a full planet import you probably used a [flatnode file](https://nominatim.org/release-docs/4.5/admin/Import/#flatnode-files). That means instead of importing OSM nodes into a `planet_osm_nodes` table that data got imported into the file (about 90GB for planet). You can double-check if you have such a table in the database. If you lost the file then you'd need to reinstall and reimport.

**Comment by RailsGuy** [comment]
Thanks but I think there is confusion because there are 2 directories named nominatim-project in the guides. One uses for imports but also another referred to in the setting-up-a-webserver guide and that is the one I deleted. 
https://nominatim.org/release-docs/latest/admin/Install-on-Ubuntu-22/#setting-up-a-webserver

**Reply by mtmail** [reply]
You got lucky then. Loosing 90GB of data would've caused a lot of extra work. Does the webserver work now? I see you opened a separate issue at https://github.com/osm-search/Nominatim/discussions/3564

**Reply by RailsGuy** [reply]
The UI is working now. I'm not sure what `$USERHOME/nominatim-project` is for other than a working dir for the gunicorn application?

**Reply by lonvia** [reply]
The project directory used to be important for running the PHP frontend because it contained the trampoline code for running the API. With the Python frontend, it's not really needed anymore. At least when you run a standard installation. If you want to customize your frontend, for example using [adapted result formatting](https://nominatim.org/release-docs/latest/customize/Result-Formatting/), then Nominatim will search for your plugin code in the project directory.
